### ISSUE TITLE
Data leakage

### ISSUE THREAD

**Issue by xbh0720** [xiaopengguo/ATKT#1]
alphas=nn.Softmax(dim=1)(att_w) 
should be masked

**Comment by lilstrawberry** [comment]
> alphas=nn.Softmax(dim=1)(att_w) should be masked

His code should have no data leakage, and his attention implementation is not a seq mask for the second dimension after it becomes seq*seq as written in the paper. After doing an mlp on seq*1, he uses cumsum to ensure that there will be no data leakage.

**Comment by lilstrawberry** [comment]
> alphas=nn.Softmax(dim=1)(att_w) should be masked

His code should have no data leakage, and his attention implementation is not a seq mask for the second dimension after it becomes seq*seq as written in the paper. After doing an mlp on seq*1, he uses cumsum to ensure that there will be no data leakage.

**Comment by lilstrawberry** [comment]
> alphas=nn.Softmax(dim=1)(att_w) should be masked

His code should have no data leakage, and his attention implementation is not a seq mask for the second dimension after it becomes seq * seq as written in the paper. After doing an mlp on seq * 1, he uses cumsum to ensure that there will be no data leakage.

**Comment by lilstrawberry** [comment]
Yes, the future information is used when calculating the attention value of each state of the sequence, resulting in inflated performance
